### DISCUSSION TITLE
esp32-s3 supported ?

### DISCUSSION THREAD

**Discussion by Wek-mad** [tobozo/WUD-Ducky#18] [Q&A]
can it work with esp32-s3 ? I'm can't find any s2 in my town. tks

**Comment by tobozo** [comment]
ESP32-S3 is not the device intended for this project but since it meets all the hardware requirements it could probably work, make sure you're using the [latest branch](https://github.com/tobozo/WUD-Ducky/tree/arduino-esp32-2.0.3-RC1)

**Comment by Wek-mad** [comment]
thanks, i will try comeback with the result.

**Comment by Wek-mad** [comment]
i tested it with the S3 but when i try to access the webpage it just a blank white page, any idea or fix ?

**Reply by tobozo** [reply]
> any idea or fix

I have a LoLin S3Pro to test with eventually, but I can reach the WebUI all right and I have no idea what to do to reproduce your blank page, so I'll need more hardware details on your build, a list of the modifications you made to the code to get it working, and a capture of your menu options (here's mine:).

![image](https://user-images.githubusercontent.com/1893754/229289150-8037560e-76ab-4db4-9510-41e35e180dbc.png)

**Reply by Wek-mad** [reply]
![image](https://user-images.githubusercontent.com/127451698/229293536-2711443e-26a6-47dd-8f09-d992dbbd7e70.png)
this is my board (ESP32-S3-WROOM-1-N16R8)

this is my menu, I actually not change anything to make the code working. 
If you need anything else can i have your discord or something? 
I would really want to use your code for my cybersecurity class.

**Reply by tobozo** [reply]
> I actually not change anything to make the code working.

then you didn't use the latest branch as it [wouldn't let you compile on esp32s3](https://github.com/tobozo/WUD-Ducky/blob/arduino-esp32-2.0.3-RC1/config_check.h#L81)

use the arduino-esp32-2.0.3-RC1 branch, then just comment out whatever line the compiler complains about and you should be good to go.

> this is my board (ESP32-S3-WROOM-1-N16R8)

oh you're using a bare module ? I'm not sure about the behaviour of the app without a SD card reader though. 

Also LittleFS doesn't cope well with USBMSC and can get corrupted, so if you're not planning to add a SD Card reader, be prepared to use the ESP32 Sketch Data Upload utility a lot!

The positive side: with your bare module you can probably listen to both uart and usb jtag, and set the `Core Debug Level` to `debug` to see what happens under the hood. 

Also if you don't have installed [this tool](https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/usbview) yet, I strongly recommend you do.

> I would really want to use your code for my cybersecurity class.

contributions to this project are welcome !

**Reply by Wek-mad** [reply]
I'm using the newest branch and it said _This project is for ESP32-S2 only._
I remove the _config_check.h_ and it compile smoothly without any error
The AP it up and running but when access the WebUI by typing 192.168.4.1 it just a blank page without anything.
I might adding a sd card module later this week or next week, It might solve the problem
